### PR TITLE
[MISC] Move print_legal function to format_help_base.

### PR DESCRIPTION
**Pull request by smehringer**
Last part of the refactoring.

This PR adds the print legal functionality to html and man output and then moves the function itself to the format_help_base to avoid inconsistencies later on.

### CODE REVIEW

File: test/unit/argument_parser/detail/format_man_test.cpp [hunk 1]
```diff
@@ -208,12 +207,11 @@ TEST_F(format_man_test, full_info_short_and_citation)
     parser.info.short_copyright = "short copyright";
     parser.info.citation = "citation";
     expected += R"(.SH LEGAL
-\fBdefault Copyright:\fR short copyright
-.br
-\fBSeqAn Copyright:\fR 2006-2015 Knut Reinert, FU-Berlin; released under the 3-clause BSDL.
+\fBdefault Copyright: \fRshort copyright
 .br
-\fBIn your academic works please cite:\fR citation
+\fBSeqAn Copyright: \fR2006-2020 Knut Reinert, FU-Berlin; released under the 3-clause BSDL.
```
Comment (eseiler) [inline test/unit/argument_parser/detail/format_man_test.cpp:212]:
```suggestion
\fBSeqAn Copyright: \fR2006-2021 Knut Reinert, FU-Berlin; released under the 3-clause BSDL.
```

File: test/unit/argument_parser/detail/format_man_test.cpp [hunk 2]
```diff
@@ -185,10 +185,9 @@ TEST_F(format_man_test, full_info_short_copyright)
     // Add a short copyright and test the dummy parser.
     parser.info.short_copyright = "short copyright";
     expected += R"(.SH LEGAL
-\fBdefault Copyright:\fR short copyright
-.br
-\fBSeqAn Copyright:\fR 2006-2015 Knut Reinert, FU-Berlin; released under the 3-clause BSDL.
+\fBdefault Copyright: \fRshort copyright
 .br
+\fBSeqAn Copyright: \fR2006-2020 Knut Reinert, FU-Berlin; released under the 3-clause BSDL.
```
Comment (eseiler) [inline test/unit/argument_parser/detail/format_man_test.cpp:190]:
```suggestion
\fBSeqAn Copyright: \fR2006-2021 Knut Reinert, FU-Berlin; released under the 3-clause BSDL.
```

File: test/unit/argument_parser/detail/format_man_test.cpp [hunk 3]
```diff
@@ -234,11 +232,11 @@ TEST_F(format_man_test, full_info_short_long_and_citation)
     parser.info.citation = "citation";
     parser.info.long_copyright = "looong copyright";
     expected += R"(.SH LEGAL
-\fBdefault Copyright:\fR short copyright
+\fBdefault Copyright: \fRshort copyright
 .br
-\fBSeqAn Copyright:\fR 2006-2015 Knut Reinert, FU-Berlin; released under the 3-clause BSDL.
+\fBSeqAn Copyright: \fR2006-2020 Knut Reinert, FU-Berlin; released under the 3-clause BSDL.
```
Comment (eseiler) [inline test/unit/argument_parser/detail/format_man_test.cpp:237]:
```suggestion
\fBSeqAn Copyright: \fR2006-2021 Knut Reinert, FU-Berlin; released under the 3-clause BSDL.
```

**Review by joergi-w — APPROVED** [review]
I have nothing to add to @eseiler's suggestions :)

**Review by eseiler — APPROVED** [review]

**Pull request merged**